### PR TITLE
replacing old converters

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: benchmarks/module/combinatorial/gcd/regular/gcd.hpp [hunk 1]
```diff
@@ -17,7 +17,7 @@ int main()
   auto arg0 = eve::bench::random_<EVE_VALUE>(lmin,lmax);
   auto arg1 = eve::bench::random_<EVE_VALUE>(lmin,lmax);
 
-  auto std__gcd =  [](auto x,  auto y){return std::gcd(eve::int_(x), eve::int_(y)); };
+  auto std__gcd =  [](auto x,  auto y){return std::gcd(eve::int_(x, to_int<decltype(x)), eve::convert(y, int_from<decltype(y))); };
```
Comment (jfalcou) [inline benchmarks/module/combinatorial/gcd/regular/gcd.hpp:20]: Missing convert here

File: include/eve/module/core/regular/converter.hpp [hunk 2]
```diff
@@ -592,9 +593,9 @@ using int_converter = decorated<convert_by_<as_integer>()>;
 //!  @godbolt{doc/core/int_.cpp}
 //!  @}
 //================================================================================================
-inline constexpr int_converter const int_ = {};
+  inline constexpr int_converter const int_ = {};
 
-using uint_converter = decorated<convert_by_<as_uinteger>()>;
+  using uint_converter = decorated<convert_by_<as_uinteger>()>;
```
Comment (jfalcou) [inline include/eve/module/core/regular/converter.hpp:598]: shouldn't this file be removed now ?

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: benchmarks/module/combinatorial/gcd/regular/gcd.hpp (hunk 1 above)
Comment (jtlap) [reply benchmarks/module/combinatorial/gcd/regular/gcd.hpp:20]: done

File: include/eve/module/core/regular/converter.hpp (hunk 2 above)
Comment (jtlap) [reply include/eve/module/core/regular/converter.hpp:598]: done

File: include/eve/module/core/regular/impl/next.hpp [hunk 3]
```diff
@@ -40,7 +39,7 @@ next_(EVE_SUPPORTS(cpu_), T const& a, U const& n) noexcept
     if constexpr( floating_value<T> )
     {
       using i_t = as_integer_t<T>;
-      return bitfloating(bitinteger(a) + to_<i_t>(n));
+      return bitfloating(bitinteger(a) + convert(as<elementype_t<i_t>>());
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/next.hpp:42]: There is something fishy here, this code is wrong yet compiles. Cna you check the include/usefulness of this file ?

File: include/eve/module/math/regular/exp2.hpp [hunk 4]
```diff
@@ -164,8 +164,8 @@ namespace eve
     exp2_(EVE_REQUIRES(cpu_), O const&, T xx, as<U> const & ) noexcept
     {
       using b_t = as_wide_as_t<U, T>;
-      using i_t  = as_integer_t<b_t>;
-      auto x  = to_<i_t>(xx);
+      using i_t  = as_integer_t<element_type_t<b_t>>;
```
Comment (jfalcou) [inline include/eve/module/math/regular/exp2.hpp:167]: isn't it int_from then ?

File: test/unit/module/core/convert/to_tuple.cpp [hunk 5]
```diff
@@ -1,66 +0,0 @@
-//==================================================================================================
```
Comment (jfalcou) [inline test/unit/module/core/convert/to_tuple.cpp:1]: this file should still be there

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: test/unit/module/core/convert/to_tuple.cpp (hunk 5 above)
Comment (jtlap) [reply test/unit/module/core/convert/to_tuple.cpp]: done

File: include/eve/module/math/regular/exp2.hpp (hunk 4 above)
Comment (jtlap) [reply include/eve/module/math/regular/exp2.hpp:167]: don't think so as b_t can be unsigned

**Review by jfalcou — APPROVED** [review]

**Review by jfalcou — APPROVED** [review]

**Pull request merged**